h/auth/util.py

```diff
@@ -117,6 +118,42 @@ def default_authority(request):
     return text_type(request.registry.settings.get('h.authority', request.domain))
 
 
+def client_authority(request):
+    """
+    Return the authority associated with an authenticated auth_client or None
+
+    Once a request with an auth_client is authenticated, a principal is set
+    indicating the auth_client's verified authority
+
+    see :func:`~h.auth.util.principals_for_auth_client` for more details on
+    principals applied when auth_clients are authenticated
+
+    :rtype: str or None
+    """
+    for principal in request.effective_principals:
+        match = re.match(r"^authority:(.+)$", principal)
```

Comment:
Regular expressions, yes, very used to. _Python_ `re`: n00bie. Would love a set of eyes on this to make sure I'm not doing anything daft.